Clarify points brought up from public consultation (#1465)
python/peps: pep-0621.rst

``[project]``. No tools may add fields to this table which are not
defined by this PEP. For tools wishing to store their own settings in
``pyproject.toml``, they may use the ``[tool]`` table as defined in

> :pep:`518`. The lack of a ``[project]`` table implicitly means all
> fields will be dynamically provided by the build tool.

Reviewer: ```suggestion
:pep:`518`. The lack of a ``[project]`` table implicitly means the
build tool will dynamically provide all fields.
```

3. If both ``email`` and ``name`` are provided, the value goes in
   ``Author-email``, with the format ``{name} <{email}>``.

This use of the field is a slight deviation as the core metadata also

> supports a ``Maintainer`` field which going unused by this PEP. The

Reviewer 2: ```suggestion
supports a ``Maintainer`` field which goes unused by this PEP. The
```

authors of this PEP felt this field was ambiguous and never clearly

> defined. Other ecosystems have selected "author" as the term so use.

Reviewer 3: "_the term **t**o use_"

> defined. Other ecosystems have selected "author" as the term so use.

Author: ```suggestion
defined. Other ecosystems have selected "author" as the term to use.
```
